.github/workflows/failed-workflow.yml

```diff
@@ -41,6 +41,22 @@ jobs:
               run_id: process.env.RUN_ID,
             });
 
+            // Output relevant URLs to workflow logs
+            console.log('=== Failed Workflow Rerun Summary ===');
+            console.log(`Workflow Run: https://github.com/${context.repo.owner}/${context.repo.repo}/actions/runs/${process.env.RUN_ID}`);
+            console.log(`Workflow Name: ${workflow_run.data.name}`);
+            console.log(`Workflow Run Attempt: ${workflow_run.data.run_attempt}`);
+            if (workflow_run.data.head_branch) {
+              console.log(`Branch: ${workflow_run.data.head_branch}`);
+            }
+            if (workflow_run.data.pull_requests && workflow_run.data.pull_requests.length > 0) {
+              console.log(`Pull Requests:`);
+              for (const pr of workflow_run.data.pull_requests) {
+                console.log(`  - https://github.com/${context.repo.owner}/${context.repo.repo}/pull/${pr.number}`);
+              }
+            }
+            console.log('====================================');
+
```

Comment:
When using the `actions/github-scripts` action, there are [3 methods available for adding annotations](https://github.com/actions/toolkit/tree/main/packages/core#annotations): `core.error()`, `core.warning()`, and `core.notice()`.

These will output information to the job log and also surface them on the workflow run summary page. I think that the `core.notice()` level annotations are appropriate here. Let's try using that method instead of direct `console.log()` calls.